Network models

ar4s opened the pull request:
Network module contains models for network and IP and basic IP Address
Management.

szok commented on src/ralph/data_center/models/networks.py:
```diff
@@ -255,196 +207,19 @@ def save(self, *args, **kwargs):
+        return ip.prefixlen
+
+    def get_subnetworks(self):
+        return self.get_descendants()
```
:+1: :+1: :+1: :+1: :+1: :+1:

mkurek commented on src/ralph/data_center/admin.py:
```diff
@@ -292,8 +295,12 @@ class DiskShareMountAdmin(RalphAdmin):
 @register(Network)
-class NetworkAdmin(RalphAdmin):
-
+class NetworkAdmin(RalphAdmin, MPTTModelAdmin):
```
take a look at https://github.com/allegro/ralph/pull/1856/files#diff-50d30b5a4f5af9a5afbb0f646b7b9ea0R308 (i hope it will be merged today)

mkurek commented on src/ralph/data_center/models/networks.py:
```diff
@@ -255,196 +214,16 @@ def save(self, *args, **kwargs):
-            parsed.append(f((self.min_ip, self.max_ip, 0)))
-        return parsed
-
     def get_netmask(self):
```
why not `netmask` property?

mkurek commented on src/ralph/data_center/models/networks.py:
```diff
@@ -255,196 +214,16 @@ def save(self, *args, **kwargs):
-            return int(mask)
-        except (ValueError, IndexError):
-            return None
+        ip = ipaddress.ip_network(self.address)
```
use `self.network.prefixlen`

mkurek commented on src/ralph/data_importer/resources.py:
```diff
@@ -180,14 +180,15 @@ class NetworkResource(RalphModelResource):
-        attribute='asset',
-        widget=ImportedForeignKeyWidget(assets.BaseObject),
+        attribute='base_object',
+        widget=ImportedForeignKeyWidget(physical.DataCenterAsset),
```
why?

szok commented on src/ralph/data_center/admin.py:
```diff
@@ -39,11 +47,55 @@
+                    parent.get_absolute_url(), parent
+                )
+            if message:
+                self.message_user(request, mark_safe(message))
```
Maybe better move under "if obj:" ?

szok commented on src/ralph/data_center/migrations/0007_auto_20160225_1818.py:
```diff
@@ -10,7 +10,6 @@ class Migration(migrations.Migration):
         ('supports', '0005_auto_20160105_1222'),
         ('assets', '0008_auto_20160122_1429'),
         ('licences', '0002_auto_20151204_1325'),
-        ('data_center', '0006_rack_require_position'),
```
why you deleted migration?

xliiv commented on src/ralph/data_center/models/networks.py:
```diff
@@ -124,41 +79,48 @@ class Meta:
+
+    def _assign_parent(self):
+        parent = getattr(self, self._parent_attr)
+        if not self.pk and not parent or self.pk:
```
This condition is not readable, imo

ar4s committed:
Added network moule

Network module contains models for network and IP and basic IP Address
Management.
